content/en/blog/_posts/2018-07-12-resizing-persistent-volumes-using-kubernetes.md

```diff
@@ -0,0 +1,82 @@
+
+![PVC StorageClass](/images/blog/2018-07-12-resizing-persistent-volumes-using-kubernetes/pvc-storageclass.png)
+
+## File System Expansion
+
+Block storage volume types such as GCE-PD, AWS-EBS, Azure Disk, Cinder, and Ceph RBD typically require a file system expansion before the additional space of an expanded volume is usable by pods. Kubernetes takes care of this automatically whenever the pod(s) referencing your volume are restarted.
+
+Network attached file systems (like Glusterfs and Azure File) can be expanded without having to restart the referencing Pod, because these systems do not require special file system expansion.
+
+File system expansion must be triggered by terminating the pod using the volume. More specifically:
+
+* Edit the PVC to request more space.
+* Once underlying volume has been expanded by the storage provider, then the PersistentVolume object will reflect the updated size and PVC will have the `FileSystemResizePending` condition.
+
+You can verify this by running `kubectl get pvc <pvc_name> -o yaml`
+
+```
+~> kubectl get pvc myclaim -o yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: myclaim
+  namespace: default
+  uid: 02d4aa83-83cd-11e8-909d-42010af00004
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 14Gi
+  storageClassName: standard
+  volumeName: pvc-xxx
+status:
+  capacity:
+    storage: 9G
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: 2018-07-11T14:51:10Z
+    message: Waiting for user to (re-)start a pod to finish file system resize of
+      volume on node.
+    status: "True"
+    type: FileSystemResizePending
+  phase: Bound
+```
+* Once PVC has the condition `FileSystemResizePending` then pod that uses the PVC can be restarted to finish file system resizing on the node. Restart can be achieved  by scaling down the deployment and scaling it up or by deleting and recreating the pod.
+* Once file system resizing is done, the PVC will automatically be updated to reflect new size.
+
+Any errors encountered while expanding file system should be available as events on pod.
+
+## Online File System Expansion
+
+Kubernetes v1.11 also introduces an alpha feature called online file system expansion.This feature enables file system expansion while a volume is still in-use by a pod. Because this feature is alpha, it requires enabling the feature gate, `ExpandInUsePersistentVolumes`. It is supported by the in-tree volume plugins GCE-PD, AWS-EBS, Cinder, and Ceph RBD. When this feature is enabled, pod referencing the resized volume do not need to be restarted, instead the file system will automatically be resized while in use as part of volume expansion. File system expansion does not happen until a pod references the resized volume, so if no pods referencing the volume are running file system expansion will not happen.
```

Comment:
1. extra space needed before `This feature`.

2. i would refactor:
> do not need to be restarted, instead the file system

to:
> do not need to be restarted. Instead, the file system

Reply:
the docs location for this are here BTW, perhaps we should reference it at the bottom of the blog post?
http://k8s.io/docs/concepts/storage/persistent-volumes